src/clang-cpp-frontend/clang_cpp_convert.cpp

```diff
@@ -826,8 +826,15 @@ bool clang_cpp_convertert::get_constructor_call(
    */
 
   // Calling base constructor from derived constructor
+  bool need_new_obj = false;
+
   if (new_expr.base_ctor_derived())
     gen_typecast_base_ctor_call(callee_decl, call, new_expr);
+  else if (
+    new_expr.get_bool("memberInit") &&
+    !callee_decl.type().return_type().get_bool("#copy_cons"))
```

Comment:
Could you add a comment indicating why copy-constructors are special here? For instance, I'm wondering what move-constructors would do, or actually in what way user-defined constructors are different.